File: MobileApp/my_social_app/lib/models/token.dart
import 'package:json_annotation/json_annotation.dart';
part 'token.g.dart';

@JsonSerializable()
class Token{
  final String accessToken;
  final DateTime expirationDateOfAccessToken;
  final String refreshToken;
  final DateTime expirationDateOfRefreshToken;

  const Token(this.accessToken,this.expirationDateOfAccessToken,this.refreshToken,this.expirationDateOfRefreshToken);

  factory Token.fromJson(Map<String, dynamic> json) => _$TokenFromJson(json);
  Map<String, dynamic> toJson() => _$TokenToJson(this);
}

File: src/MySocailApp.Domain/AccountAggregate/Token.cs
namespace MySocailApp.Domain.AccountAggregate
{
    public class Token(string accessToken, DateTime expirationDateOfAccessToken, string refreshToken, DateTime expirationDateOfRefreshToken)
    {
        public string AccessToken { get; private set; } = accessToken;
        public DateTime ExpirationDateOfAccessToken { get; private set; } = expirationDateOfAccessToken;
        public string RefreshToken { get; private set; } = refreshToken;
        public DateTime ExpirationDateOfRefreshToken { get; private set; } = expirationDateOfRefreshToken;
    }
}
